composer/trainer/trainer.py

```diff
@@ -2980,7 +2991,7 @@ def _eval_loop(
                                     rank_num_samples -= 1
                                     num_samples_in_microbatch = data_spec.get_num_samples_in_batch(self.state.batch)
                                     # Skip updating metric if batch is only padded samples
-                                    if num_samples_in_microbatch == 1:
+                                    if num_samples_in_microbatch <= 1 or rank_num_samples <= 0:
```

Comment:
@mvpatel2000 . In the case of sequence parallelism, rank_num_samples (per gpu) will be less than or equal to 1 (at least right now, until we enable that). So, in that case, is the modification correct? I think for sequence parallelism, the following lines (which are a few lines later) should never be executed.
`                                        self.state.batch = data_spec.split_batch(self.state.batch,
                                                                                 num_samples_in_microbatch - 1)[0]`